qforce/qm/xtb.py

```diff
@@ -0,0 +1,452 @@
+    _user_input = """
+    charge_method = xtb :: str ::
+    """
+
+    _method = []
+
+    def __init__(self):
+        self.required_hessian_files = {'hess_file': ['hessian'],
+                                       'pc_file': ['charges'],
+                                       'wbo_file': ['wbo'],
+                                       'coord_file': ['xtbopt.xyz'], }
+        self.read = ReadxTB
+        self.write = WritexTB
+
+class WritexTB(WriteABC):
+    def hessian(self, file, job_name, config, coords, atnums):
+        """ Write the input file for hessian and charge calculation.
+
+        Parameters
+        ----------
+        file : file
+            The file object to write the command line.
+        job_name : string
+            The name of the job.
+        config : config
+            A configparser object with all the parameters.
+        coords : array
+            A coordinates array of shape (N,3), where N is the number of atoms.
+        atnums : list
+            A list of atom elements represented as atomic number.
+        """
+        name = file.name
+        base, filename = os.path.split(name)
+        # Given that the xTB input has to be given in the command line.
+        # We create the xTB command template here.
+        cmd = f'xtb {job_name}_input.xyz --ohess --chrg {config.charge} ' \
+              f'--uhf {config.multiplicity - 1} ' \
+              f'--namespace {job_name}_hessian --parallel {config.n_proc}'
+        # Write the hessian.inp which is the command line input
+        file.write(cmd)
+        # Write the coordinates, which is the standard xyz file.
+        with open(f'{base}/{job_name}_input.xyz', 'w') as f:
+            f.write(f'{len(atnums)}\n{cmd}\n')
+            self._write_coords(atnums, coords, f)
+
+    def scan(self, file, job_name, config, coords, atnums, scanned_atoms,
+             start_angle, charge, multiplicity):
+        """ Write the input file for the dihedral scan and charge calculation.
+
+        Parameters
+        ----------
+        file : file
+            The file object to write the input.
+        job_name : string
+            The name of the job.
+        config : config
+            A configparser object with all the parameters.
+        coords : array
+            A coordinates array of shape (N,3), where N is the number of atoms.
+        atnums : list
+            A list of atom elements represented as atomic number.
+        scanned_atoms : list
+            A list of 4 integer (one-based index) of the dihedral
+        start_angle : float
+            The starting angle in degree.
+        charge : int
+            The total charge of the molecule.
+        multiplicity : int
+            The multiplicity of the molecule.
+        """
+        # Write the xTB input file
+        name = file.name
+        base, filename = os.path.split(name)
+
+        # Generate the command line
+        cmd = f'xtb {job_name}_input.xyz --opt --chrg {config.charge} ' \
```

Comment:
same as before, these settings can be provided in the input file that you already have